WordPress/src/jetpack/assets/support_chat_widget.js

```diff
@@ -1,31 +1,109 @@
-        var n = document.getElementsByTagName('script')[0]
-        n.parentNode.insertBefore(t, n),
-          t.addEventListener('load', function () {
-            window.DocsBotAI.mount({
-              id: c.id,
-              supportCallback: c.supportCallback,
-              identify: c.identify,
-              options: c.options,
-            })
-            var t
-            ;(t = function (n) {
-              return new Promise(function (e) {
-                if (document.querySelector(n)) return e(document.querySelector(n))
-                var o = new MutationObserver(function (t) {
-                  document.querySelector(n) && (e(document.querySelector(n)), o.disconnect())
-                })
-                o.observe(document.body, { childList: !0, subtree: !0 })
-              })
-            }),
-              t && t('#docsbotai-root').then(e).catch(o)
-          }),
-          t.addEventListener('error', function (t) {
-            o(t.message)
-          })
-      })
-    })
+( function () {
+
+    function init() {
+        loadDocsBotAI()
+    }
+
+    function loadDocsBotAI() {
+        const urlParams = getQueryParams(window.location);
+
+        window.DocsBotAI.init({
+            id: urlParams.id,
+            supportCallback: function(event, history) {
+                event.preventDefault() // Optionally prevent default behavior opening the url.
+                console.log(history) // Safely access the chat history.
+                DocsBotAI.close() // Close the widget.
+                sendAndroidMessage(history)
+            },
+            options: {
+                botName: "Jetpack Mobile",
+                description: "Ask me anything about using Jetpack mobile app.",
+                color: "#9dd977",
+                supportLink: "#",
+                labels: {
+                    inputPlaceholder: decodeURIComponent(urlParams.inputPlaceholder),
+                    firstMessage: decodeURIComponent(urlParams.firstMessage),
+                    sources: "Sources",
+                    helpful: "Rate as helpful",
+                    unhelpful: "Rate as unhelpful",
+                    getSupport: decodeURIComponent(urlParams.getSupport),
+                    floatingButton: "Help",
+                    suggestions: decodeURIComponent(urlParams.suggestions),
+                    thinking: "Thinking..."
+                }, // Override all the default labels for your own language.
+                questions: [
+                    decodeURIComponent(urlParams.questionOne),
+                    decodeURIComponent(urlParams.questionTwo),
+                    decodeURIComponent(urlParams.questionThree)
+                ] // Array of example questions to show in the widget. Three are picked at random.
+            },
+        }).then(() => {
+            // Safely do stuff here after the widget is loaded.
+            let timerId = setInterval(() => {
+                openDocsBot(); // wait for init
```

Comment:
This method does not always work, perhaps a `200` interval is not always sufficient. During testing on both iOS and Android, I have observed that sometimes the chat appears as expected, while other times we need to click on the chat icon.

Reply:
Probably, need to look at with the help of a JS expert or DocsBot people to open straight away instead of a widget, and click/open mechanism.

Reply:
True. I'm looking at it but it's hard to say why it doesn't work sometimes. `.init` fulfills a promise when `docsbotai-root` element is found. It makes sense that at that moment we should be able to open the chat. I tried other approaches, such as observing `#docsbotai-root` myself. It seems to be more reliable although I cannot find arguments why it's more reliable.